src/lib.rs

```diff
@@ -1604,6 +1602,14 @@ impl<T> Drop for InactiveReceiver<T> {
     fn drop(&mut self) {
         if let Ok(mut inner) = self.inner.lock() {
             inner.inactive_receiver_count -= 1;
+
+            close_channel_if_last_receiver(&mut inner);
         }
     }
 }
+
+fn close_channel_if_last_receiver<T>(inner: &mut MutexGuard<'_, Inner<T>>) {
```

Comment:
This should be a method of `Inner<T>` that takes a `&mut self`.

Reply:
makes sense